api/cloudcontroller/ccv3/metadata.go

```diff
@@ -1,10 +1,67 @@
 package ccv3
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
+
+	"code.cloudfoundry.org/cli/api/cloudcontroller"
+	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv3/internal"
 	"code.cloudfoundry.org/cli/types"
 )
 
 // Metadata is used for custom tagging of API resources
 type Metadata struct {
 	Labels map[string]types.NullString `json:"labels,omitempty"`
 }
+
+type ResourceMetadata struct {
+	Metadata *Metadata `json:"metadata,omitempty"`
+}
+
+func (client *Client) UpdateResourceMetadata(resource string, resourceGUID string, metadata Metadata) (ResourceMetadata, Warnings, error) {
+	metadataBtyes, err := json.Marshal(ResourceMetadata{Metadata: &metadata})
+	if err != nil {
+		return ResourceMetadata{}, nil, err
+	}
+
+	var request *cloudcontroller.Request
+
+	switch resource {
+	case "app":
```

Comment:
All of these tests just switch on endpoint and the URI Params. I wasn't sure what the most idiomatic way to do this was in Go so would appreciate any suggestions here. 🙏